hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+        energy_Mn_Ka, FWHM_MnKa_old = self.spectrum._get_line_energy('Mn_Ka',
+                                                                     'auto')
+        FWHM_MnKa_old *= eV2keV / self.units_factor
+        get_sigma_Mn_Ka = _get_sigma(
+            energy_Mn_Ka, self[xray_lines[0]].centre.value, self.units_factor)
+        FWHM_MnKa = get_sigma_Mn_Ka(self[xray_lines[0]].sigma.value
+                                    ) * eV2keV / self.units_factor * sigma2fwhm
+        if FWHM_MnKa < 110:
+            warnings.warn("FWHM_MnKa of " + str(FWHM_MnKa) + " smaller than" +
+                          "physically possible")
+        else:
+            self.spectrum.set_microscope_parameters(
+                energy_resolution_MnKa=FWHM_MnKa)
+            warnings.warn("Energy resolution (FWHM at Mn Ka) changed from "
+                          + "%lf to %lf eV" % (FWHM_MnKa_old, FWHM_MnKa))
+            for component in self:
+                if component.isbackground is False:
+                    line_energy, line_FWHM = self.spectrum._get_line_energy(
+                        component.name, FWHM_MnKa='auto')
+                    component.fwhm = line_FWHM
+
+    def _twin_xray_lines_scale(self, xray_lines):
+        """
+        Twin the scale of the peaks
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+        ax = self.spectrum.axes_manager[-1]
+        ref = []
+        for i, xray_line in enumerate(xray_lines):
+            component = self[xray_line]
+            if i == 0:
+                component_ref = component
+                component_ref.centre.free = True
+                E_ref = component_ref.centre.value
+                ref.append(E_ref)
+            else:
+                component.centre.free = True
+                E = component.centre.value
+                fact = float(ax.value2index(E)) / ax.value2index(E_ref)
+                component.centre.twin_function = _get_scale(E, E_ref, fact)
+                component.centre.twin_inverse_function = _get_scale(
+                    E_ref, E, 1./fact)
+                component.centre.twin = component_ref.centre
+                ref.append(E)
+        return ref
+
+    def _set_energy_scale(self, xray_lines, ref):
+        """
+        Adjust the width of all lines and set the fitted energy resolution
+        to the spectrum
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        ref: float
+            the centre of the first line before the fit
```

Comment:
Documentation is wrong - in the code `ref` is not just one number, but as long as the `xray_lines` are (and of 'all_alpha', then as many components the model has)